doc/source/tutorial/stats/outliers.md

```diff
@@ -0,0 +1,521 @@
+
+```{code-cell} ipython3
+mxp.mean(x2d_masked, axis=-1)
+```
+
+Many SciPy functions also natively support masked arrays (see the documentation of the desired function).
+
+```{code-cell} ipython3
+stats.kurtosis(x2d_masked, axis=-1)
+```
+
+When the trimmed values are represented with NaNs, there are some NumPy functions that ignore NaNs:
+
+```{code-cell} ipython3
+np.nanmean(x2d_nan, axis=-1)
+```
+
+And many SciPy functions that accept a `nan_policy` argument.
+
+```{code-cell} ipython3
+stats.kurtosis(x2d_nan, nan_policy='omit', axis=-1)
+```
+
+# Recipes
+
++++
+
+Here, we reproduce the results of several legacy trimming functions using the preferred approach.
+
+```{code-cell} ipython3
+rng = np.random.default_rng(339759635264198322188079188741355346875)
+x = rng.random(22)
+p = 0.1
+```
+
+## `stats.trim_mean`
+
+```{code-cell} ipython3
+stats.trim_mean(x, p)
+```
+
+```{code-cell} ipython3
+low, high = stats.quantile(x, [p, 1-p], method='trim_less')
+mask = (low < x) & (x < high)
+np.mean(x[mask])
+```
+
+## `stats.tmean`
+
+```{code-cell} ipython3
+limits = stats.quantile(x, [p, 1-p], method='winsor_less')
+```
+
+```{code-cell} ipython3
+stats.tmean(x, limits, inclusive=(True, True))
+```
+
+```{code-cell} ipython3
+mask = (limits[0] <= x) & (x <= limits[1])
+np.mean(x[mask])
+```
+
+```{code-cell} ipython3
+stats.tmean(x, limits, inclusive=(False, False))
+```
+
+```{code-cell} ipython3
+mask = (limits[0] < x) & (x < limits[1])
+np.mean(x[mask])
+```
+
+## `stats.mstats.trimmed_mean`
+
+```{code-cell} ipython3
+stats.mstats.trimmed_mean(x, (p, p), inclusive=(True, True), relative=True)
+```
+
+Surprisingly, `inclusive` has no effect when `p*len(x)` is integral and `relative` is `True`.
+
+```{code-cell} ipython3
+stats.mstats.trimmed_mean(x, (p, p), inclusive=(False, False), relative=True)
+```
+
+So let's consider a case in which it would matter.
+
+```{code-cell} ipython3
+x = rng.random(size=19)
+```
+
+```{code-cell} ipython3
+stats.mstats.trimmed_mean(x, (p, p), inclusive=(True, True), relative=True)
+```
+
+```{code-cell} ipython3
+low, high = stats.quantile(x, [p, 1-p], method='winsor_less')
+mask = (low <= x) & (x <= high)
+np.mean(x[mask])
+```
+
+```{code-cell} ipython3
+stats.mstats.trimmed_mean(x, (p, p), inclusive=(False, False), relative=True)
+```
+
+```{code-cell} ipython3
+low, high = stats.quantile(x, [p, 1-p], method='winsor_less')
+mask = (low < x) & (x < high)
+np.mean(x[mask])
+```
+
+`stats.mstats.trimmed_mean` is a convenience function equivalent to:
+
+```python3
+stats.mstatats.trim(a, limits=limits, inclusive=inclusive, relative=relative).mean(axis=axis)
+```
+
+So the behavior is best understood in terms of `stats.mstats.trim`.
+
++++
+
+## `stats.mstats.trim`
```

Comment:
Oops. Forgot to mention that `stats.mstats.trimboth` and `stats.mstats.trimtail` are two more convenience wrappers around `trimr`, and `stats.trimboth` and `stats.trim1` are additional options for 1-D data that actually remove elements rather than masking them.